nucleus/core/kernel/src/main/java/com/sun/enterprise/v3/admin/CreateSystemProperties.java

```diff
@@ -154,6 +159,15 @@ public void execute(AdminCommandContext context) {
                 } finally {
                     TranslatedConfigView.doSubstitution.set(Boolean.TRUE);
                 }
+
+                Matcher matcher = HTML_JS_PATTERN.matcher(properties.getProperty(propName));
+                if (matcher.find()) {
+                    report.setMessage(localStrings.getLocalString("create.system.properties.failed",
+                            "System property {0} creation failed", sysPropName));
+                    report.setActionExitCode(ActionReport.ExitCode.FAILURE);
+                    return;
```

Comment:
Wouldn't it be clearer in terms of code structure to throw an IllegalArgumentException instead of returning directly out of the loop? This way you could have a more specific error message and leave the proper error handling to the catch below.